avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/serialization/AVRO_1_8/Array_of_LONG_GenericSerializer_2055015354772118358.java

```diff
@@ -0,0 +1,30 @@
+
+package com.linkedin.avro.fastserde.generated.serialization.AVRO_1_8;
+
+import java.io.IOException;
+import java.util.List;
+import com.linkedin.avro.fastserde.FastSerializer;
+import org.apache.avro.io.Encoder;
+
+public class Array_of_LONG_GenericSerializer_2055015354772118358
+    implements FastSerializer<List<Long>>
+{
+
+
+    public void serialize(List<Long> data, Encoder encoder)
+        throws IOException
+    {
+        (encoder).writeArrayStart();
+        if ((data == null)||data.isEmpty()) {
+            (encoder).setItemCount(0);
+        } else {
+            (encoder).setItemCount(data.size());
+            for (int counter0 = 0; (counter0 <((List<Long> ) data).size()); counter0 ++) {
```

Comment:
why are you generating a cast here ?
also, can grab size() just once

Reply:
My goal is to not modify the code-gen as part of this PR, but only to increase test coverage. Once that's in, then I will submit another PR to alter the code-gen, and show the diff in the generated code.